controller/linodecluster_controller.go

```diff
@@ -117,24 +117,51 @@ func (r *LinodeClusterReconciler) reconcile(
 	// Always close the scope when exiting this function so we can persist any LinodeCluster changes.
 	defer func() {
 		// Filter out any IsNotFound message since client.IgnoreNotFound does not handle aggregate errors
-		if err := clusterScope.Close(ctx); utilerrors.FilterOut(err, apierrors.IsNotFound) != nil && reterr == nil {
+		if err := clusterScope.Close(ctx); utilerrors.FilterOut(util.UnwrapError(err), apierrors.IsNotFound) != nil && reterr == nil {
 			logger.Error(err, "failed to patch LinodeCluster")
 			reterr = err
 		}
 	}()
 
 	// Handle deleted clusters
 	if !clusterScope.LinodeCluster.DeletionTimestamp.IsZero() {
-		return res, r.reconcileDelete(ctx, logger, clusterScope)
+		err := r.reconcileDelete(ctx, logger, clusterScope)
+		if err != nil && !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
```

Comment:
Only one condition severity check is needed here. Also, we should only requeue when the condition severity is warning, not error (which we consider terminal).

Reply:
oops. Thanks for catching that!